### DISCUSSION TITLE
EnsureSpace Not Always working, and how to Wrap text to a side column?

### DISCUSSION THREAD

**Discussion by Guilherme-Cassel** [QuestPDF/QuestPDF#758] [Q&A] [answered]
I'm doing a order report for my company, we project and manufacture machines.

On This Report I have a Header with info, and the Order Items Stacked one down another. 
On each order item, there is this big text box below that is the item narrative, which can vary from one sentece up to 10k characters.

But As I've been doing it, I found some lacks:

Because the narrative is so big, I wanted to split it into 2 columns, side by side, inside this big text container, the problem is that, there is no native function to make the text automatically break to the other column, so I had to split the narrative string into 2 pieces by code, and set the text to 2 different columns by hand, which is another problem because dont matter what I do, both sides will never have the same character amout because some sentences in the narrative are bigger than others.

Now comes the bug:

I'm using the EnsureSpace() extension in almost all the containers of the order item and its still pagebreaking the container sometimes, why?

![EnsureSpace Fail Example](https://github.com/QuestPDF/QuestPDF/assets/150675837/61f66bfb-dacb-4425-9187-7313d899dd14)

You Can see on the image above that the item 19 pagebroke to the next page, even tough EnsureSpace() is used on the container.

heres the code that applies the OrderItem container (TableItem):

![The Code](https://github.com/QuestPDF/QuestPDF/assets/150675837/9655648f-ff39-4043-b3f2-ed34dac1785b)

The TableItem method just divides the container into the correct columns and split the narrative and apply to column[0] and column[1].

So, resuming: 

Is There a way to wrap a big text into two columns automatically?
and is there a way to prevent EnsureSpace() failing, or is my code so bad that the EnsureSpace bugs?

Thanks for your time.

**Comment by MarcinZiabek** [comment, marked as answer]
Hi 😀

Unfortunately, splitting content between two columns is not supported at this moment. However, there is a promising prototype already (https://github.com/QuestPDF/QuestPDF/tree/feat-multicolumn-prototype). I am hoping to investigate its implementation once this https://github.com/QuestPDF/QuestPDF/discussions/622 effort is completed.

If you want to prevent your content from paging, please use the `ShowEntire` element - make sure your content can always fit within one page. Otherwise, you will get the layout exception. 

The `EnsureSpace` element is a looser version of the `ShowEntire` approach - it accepts a value in points that specifies how much space should be provided on the first page. You can increase that value higher than the default.

**Reply by Guilherme-Cassel** [reply]
Thanks, the ShowEntire() approach solved the problem.

Keep up the good work!! Thanks for your amazing job.

**Reply by MarcinZiabek** [reply]
Would you have any suggestions on how to improve API naming to make it more discoverable? Certain concepts are quite simple, yet surprisingly difficult to describe 😀

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The functionality you described is now available through the [MultiColumn](https://www.questpdf.com/api-reference/multi-column.html) element, introduced in version 2024.7.0.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
